src/main/java/seedu/address/logic/parser/EditCommandParser.java

```diff
@@ -39,53 +43,57 @@ public EditCommand parse(String args) throws ParseException {
             throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, EditCommand.MESSAGE_USAGE));
         }
     }
 
-    private EditCommand handleTutor(String args) throws ParseException {
-        ArgumentMultimap tutorArgMultimap =
-                ArgumentTokenizer.tokenize(args, PREFIX_NAME, PREFIX_PHONE, PREFIX_GENDER,
-                        PREFIX_QUALIFICATION, PREFIX_REMARK, PREFIX_TAG);
-
-        EditTutorDescriptor editTutorDescriptor = new EditTutorDescriptor();
-        return parsePerson(tutorArgMultimap, editTutorDescriptor, PersonType.TUTOR);
-    }
-
-    private EditCommand handleStudent(String args) throws ParseException {
-        ArgumentMultimap studentMultimap =
-                ArgumentTokenizer.tokenize(args, PREFIX_NAME, PREFIX_PHONE, PREFIX_GENDER, PREFIX_REMARK, PREFIX_TAG);
-
-        EditStudentDescriptor editStudentDescriptor = new EditStudentDescriptor();
-        return parsePerson(studentMultimap, editStudentDescriptor, PersonType.STUDENT);
-    }
-
     private EditCommand parsePerson(ArgumentMultimap argMultimap, EditPersonDescriptor editPersonDescriptor,
                                     PersonType personType) throws ParseException {
         Index personIndex;
 
         try {
             String trimmed = argMultimap.getPreamble().trim();
-            String[] split = trimmed.split(" ", 2);
+            String[] split = trimmed.split(" ");
             // Preamble should only have personType and Index
             if (split.length != 2) {
                 throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, EditCommand.MESSAGE_USAGE));
             }
-            // If personType or index not given, it is an invalid command format. ArrayIndexOutOfBoundsException
-            // will be thrown by these lines below, but we catch it and throw a ParseException
+            // If personType or index not given, it is an invalid command format. ParseException with
+            // MESSAGE_INVALID_INDEX will be thrown by the line below, but we catch it and throw a more specific and
+            // non-technical ParseException below.
             personIndex = ParserUtil.parseIndex(split[1]);
         } catch (ParseException pe) {
-            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, EditCommand.MESSAGE_USAGE), pe);
+            if (pe.getMessage().equals(MESSAGE_INVALID_INDEX)) {
+                if (personType.equals(PersonType.STUDENT)) {
+                    throw new ParseException(MESSAGE_INVALID_STUDENT_DISPLAYED_INDEX);
+                } else {
+                    throw new ParseException(MESSAGE_INVALID_TUTOR_DISPLAYED_INDEX);
+                }
+            } else {
+                throw pe;
+            }
+        }
+        if (argMultimap.getValue(PREFIX_QUALIFICATION).isPresent()) {
+            if (personType.equals(PersonType.STUDENT)) {
+                throw new ParseException(MESSAGE_INVALID_INPUT_STUDENT_WITH_QUALIFICATION);
```

Comment:
Perhaps you could put the code from here to the end of the method into another method like a `checkParametersPresent` method? LOC for the whole `parsePerson` method seems pretty long now